### PR TITLE
Add missing tests for ML APIs (search and delete)

### PR DESCRIPTION
**Pull request by nathaliellenaa**
### Description
Added missing tests for `GET /_plugins/_ml/models/_search` and `DELETE /_plugins/_ml/tasks/{task_id}`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/schemas/ml._common.yaml [hunk 1]
```diff
@@ -22,6 +22,28 @@ components:
...
       type: object
       properties:
+        took:
+          type: integer
+          description: The time taken to execute the search.
+        timed_out:
+          type: boolean
+          description: Whether the search timed out.
+        _shards:
+          type: object
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:32]: This should be a new `SearchModelsShardsResponse` or just `Shards` object.

File: spec/schemas/ml._common.yaml [hunk 2]
```diff
@@ -59,12 +84,74 @@ components:
         _id:
           type: string
           description: The hit ID.
+        _version:
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:87]: I believe this is shared from a base object, so it should become an `allOf:` with a base response that contains `_seq_no` and `_primary_term` and then the fields that are specific to this object here.

File: tests/plugins/ml/ml/models.yaml [hunk 3]
```diff
@@ -37,6 +54,14 @@ chapters:
     retry:
       count: 5
       wait: 30000
+  - synopsis: Delete task.
```
Comment (dblock) [inline tests/plugins/ml/ml/models.yaml:57]: Should be its own test in `tests/plugins/ml/ml/tasks.yaml`.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good, needs a bit of work, iterate to green.

1. Fix validation/CI failures.
2. Sign DCO, `git commit --amend -s` for existing commits.

File: spec/schemas/ml._common.yaml [hunk 4]
```diff
@@ -51,20 +89,75 @@ components:
         - relation
         - value
     SearchModelsHitsHit:
+      allOf:
+        - $ref: '#/components/schemas/BaseResponse'
+        - type: object
+          properties:
+            _index:
+              type: string
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:97]:
```
                $ref: '../schemas/_common.yaml#/components/schemas/IndexName'
```

File: spec/schemas/ml._common.yaml [hunk 5]
```diff
@@ -51,20 +89,75 @@ components:
...
+              description: The index.
+            _id:
+              type: string
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:100]:
```
                $ref: '../schemas/_common.yaml#/components/schemas/Id'
```

File: spec/schemas/ml._common.yaml [hunk 6]
```diff
@@ -51,20 +89,75 @@ components:
...
+              description: The hit ID.
+            _score:
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:102]: Add `format: float`.

File: spec/schemas/ml._common.yaml [hunk 7]
```diff
@@ -51,20 +89,75 @@ components:
...
+              type: number
+              description: The score.
+            _source:
+              $ref: '#/components/schemas/Source'
+            model_id:
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:107]: Possibly

```
            model_id:
              $ref: '_common.yaml#/components/schemas/Name'
```

File: spec/schemas/ml._common.yaml [hunk 8]
```diff
@@ -51,20 +89,75 @@ components:
...
-        _index:
+        last_deployed_time:
+          type: number
+          description: The last deployed time.
+        model_version:
           type: string
-          description: The index.
-        _id:
+          description: The model version.
+        created_time:
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:121]:
```
        created_time:
          type: integer
          format: int64
```

File: spec/schemas/ml._common.yaml [hunk 9]
```diff
@@ -51,20 +89,75 @@ components:
...
+          description: Whether to deploy to all nodes.
+        is_hidden:
+          type: boolean
+          description: Whether the model is hidden.
+        description:
           type: string
-          description: The hit ID.
-        model_id:
+          description: The model description.
+        model_state:
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:133]: Is this an enum?

File: spec/schemas/ml._common.yaml [hunk 10]
```diff
@@ -51,20 +89,75 @@ components:
...
           type: string
-          description: The model ID.
-      required:
-        - _id
-        - model_id
+          description: The model state.
+        planning_worker_node_count:
+          type: number
+          description: The planning worker node count.
+        last_updated_time:
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:139]: Is this in ms?

```
        last_updated_time:
          type: integer
          format: int64
```

**Comment by dblock** [comment]
The test failure looks legit, probably needs a refresh or maybe something more.

```
ERROR   ml/models.yaml (tests/plugins/ml/ml/models.yaml)
    ERROR   CHAPTERS
        PASSED  Register model.
            PASSED  REQUEST BODY
            PASSED  RESPONSE STATUS
            PASSED  RESPONSE PAYLOAD BODY
            PASSED  RESPONSE PAYLOAD SCHEMA
            PASSED  RESPONSE OUTPUT VALUES
        ERROR   Search model.
            PASSED  REQUEST BODY
            PASSED  RESPONSE STATUS
            FAILED  RESPONSE PAYLOAD BODY (missing hits.hits[0]='[object Object]')
            PASSED  RESPONSE PAYLOAD SCHEMA
            ERROR   RESPONSE OUTPUT VALUES (Expected to find non undefined value at `payload.hits.hits[0]._id`.)
        SKIPPED Wait to get completed task.
        SKIPPED Delete model.
```

File: spec/schemas/ml._common.yaml (hunk 9 above)
Comment (nathaliellenaa) [reply spec/schemas/ml._common.yaml:133]: Right, this is an enum

File: spec/schemas/ml._common.yaml (hunk 10 above)
Comment (nathaliellenaa) [reply spec/schemas/ml._common.yaml:139]: I believe that is in ms.

File: spec/schemas/ml._common.yaml (hunk 9 above)
Comment (dblock) [reply spec/schemas/ml._common.yaml:133]: Enumerate the values if it's a finite list.

File: spec/schemas/ml._common.yaml [hunk 11]
```diff
@@ -53,18 +66,103 @@ components:
...
+          type: boolean
+          description: Whether the model is hidden.
+        description:
+          type: string
+          description: The model description.
+        model_state:
+          type: string
+          description: The model state.
+          enum:
+            - DEPLOY_FAILED
```
Comment (dblock) [inline spec/schemas/ml._common.yaml:117]: Looks like there are many other model states possible: https://github.com/opensearch-project/ml-commons/blob/45ff4f59ac40d1b3f725effc4d3e79ff4f39d24f/common/src/main/java/org/opensearch/ml/common/model/MLModelState.java#L8

**Review by dblock — CHANGES_REQUESTED** [review]
Some tests are still failing.

File: tests/plugins/ml/ml/tasks.yaml [hunk 12]
```diff
@@ -0,0 +1,32 @@
+$schema: ../../../../json_schemas/test_story.schema.yaml
+
+description: Test the creation of tasks.
+distributions:
+  excluded:
+    - amazon-managed
+    - amazon-serverless
```
Comment (dblock) [inline tests/plugins/ml/ml/tasks.yaml:7]: Do you know for sure this API is not available in AOS? If not you can remove this for now.

**Review by dblock — CHANGES_REQUESTED** [review]
Getting there!

**Comment by dblock** [comment]
```
            FAILED  RESPONSE PAYLOAD SCHEMA (data/hits/hits/27/_source/model_content_hash_value property is not defined in the spec, data/hits/hits/27/_source/last_registered_time property is not defined in the spec, data/hits/hits/27/_source/model_content_size_in_bytes property is not defined in the, ...)
```

Looks like more fields are missing, https://github.com/opensearch-project/ml-commons/blob/9c1b8a8c1e28ab8bdaf507cb71799c4809c3d8c6/plugin/src/test/resources/org/opensearch/ml/autoredeploy/ModelResult.json#L7.

**Review by dblock — APPROVED** [review]
Passing tests!

**Pull request merged**